engine/src/main/java/io/zeebe/engine/nwe/behavior/BpmnEventPublicationBehavior.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.nwe.behavior;
+
+import io.zeebe.engine.processor.TypedStreamWriter;
+import io.zeebe.engine.processor.workflow.handlers.endevent.ErrorEventHandler;
+import io.zeebe.engine.state.ZeebeState;
+import io.zeebe.engine.state.instance.ElementInstance;
+import io.zeebe.engine.state.instance.ElementInstanceState;
+import org.agrona.DirectBuffer;
+
+public final class BpmnEventPublicationBehavior {
+
+  private final ErrorEventHandler errorEventHandler;
+  private final TypedStreamWriter streamWriter;
+  private final ElementInstanceState elementInstanceState;
+
+  public BpmnEventPublicationBehavior(
+      final ZeebeState zeebeState, final TypedStreamWriter streamWriter) {
+    final var workflowState = zeebeState.getWorkflowState();
+    final var keyGenerator = zeebeState.getKeyGenerator();
+    elementInstanceState = workflowState.getElementInstanceState();
+    errorEventHandler = new ErrorEventHandler(workflowState, keyGenerator);
+    this.streamWriter = streamWriter;
+  }
+
+  /**
+   * Throws an error event that must be caught somewhere in the scope hierarchy.
+   *
+   * @return {@code true} if the error event is thrown and caught by an catch event
+   * @see ErrorEventHandler#throwErrorEvent(DirectBuffer, ElementInstance, TypedStreamWriter)
+   */
+  public boolean throwErrorEvent(final DirectBuffer errorCode, final long flowScopeKey) {
+    final var flowScopeInstance = elementInstanceState.getInstance(flowScopeKey);
+    return errorEventHandler.throwErrorEvent(errorCode, flowScopeInstance, streamWriter);
+  }
```

Comment:
I would prefer to pass the whole context in like we do in other places. I think that the context is safer than passing just a long value in it.